### DISCUSSION TITLE
Rest API on VALETUDO RE

### DISCUSSION THREAD

**Discussion by dgel27** [rand256/valetudo#523] [Q&A] [answered]
Hi, is it present REST API.
I just tried same curl links from valetudo, and it is doesn't works.
How can i control robot with curl?
Thanks!

**Comment by rand256** [comment, marked as answer]
https://github.com/rand256/valetudo/wiki/REST-API

This API should work.

**Reply by dgel27** [reply]
Hi @rand256 , Thanks for quick answer.
<strike>Could you print here example how to pass parameters to go_to?
The "drive_home" works for me, but "go_to" give me "coordinates missing".</strike>
`
curl -X 'PUT' 'http://<IP>/api/go_to' -H 'accept: application/json' -H 'Content-Type: application/json' -d '{"x": 26477, "y": 24849}' 
`


And maybe you know:
1. is map distance on S5 limited to +- 25.5 meters from base (i cannot send it far or create map, when map coordinate less 0)
2. is it possible to add "go_to" to timing - _it possible to add via mirobo, or edit apptimer.cfg_
3. is it possible to add comand instead of voice "reach destination", or is it possible to get status when it reach the destination?

I will try to add it by myself, just could you point where it is in the code? We try project for robot operation, not for cleaning.

Thanks!

**Reply by rand256** [reply]
Well. It's been a while since the last time I wrote the code for valetudo... Let's see if I could remember something.

1. Actually, I can't say anything certain about it. But after all it's very likely the device has some limitations like you mention at its low level firmware which probably can't be altered.
2. At some point I thought to add goto into timers section the same way it is currently done for zone cleaning (which is managed solely by valetudo, no apptimer.cfg or anything involved). And basically it could be rather easy to do so in code when you already have the timed zone cleaning as a reference. But... in the end, it didn't happen. And yeah, I had no idea it was possible to implement this with the default `set_timer` miio command.
3. For this, take a look [here](https://github.com/rand256/valetudo/blob/testing/lib/miio/Vacuum.js#L193) - this code runs every time when device reaches the destination after any goto command. And [this](https://github.com/rand256/valetudo/blob/testing/lib/miio/Dummycloud.js#L76) is where it comes from.

**Reply by dgel27** [reply]
Thanks a lot for a help!
